csharp/src/Apache.Arrow/Arrays/ListArray.cs

```diff
@@ -32,14 +32,14 @@ public class ListArray : Array
             : this(new ArrayData(dataType, length, nullCount, offset,
                 new[] {nullBitmapBuffer, valueOffsetsBuffer}, new[] {values.Data}))
         {
-            Values = values;
```

Comment:
I think the existing behavior is better here - if I construct a `ListArray` instance with a specific `IArrowArray values`, I would expect the new `ListArray.Values` to return the same instance.

There isn't a reason to create a new ArrowArray (which happens in the below constructor's `ArrowArrayFactory.BuildArray` call) when this constructor is already given the `IArrowArray`.

Maybe we can split out a private helper "Initialize" method that both can call instead of having one constructor call the other?

See [https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/constructor](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/constructor)

✓ DO use constructor parameters as shortcuts for setting main properties.

There should be no difference in semantics between using the empty constructor followed by some property sets and using a constructor with multiple arguments.

✓ DO use the same name for constructor parameters and a property if the constructor parameters are used to simply set the property.

The only difference between such parameters and the properties should be casing.

Reply:
Okey.

Instead of creating an initialize method, How about changing the `ListArray(ArrayData data)` constructor as follows?

```
public ListArray(ArrayData data , IArrowArray values = null)
            : base(data)
        {
            data.EnsureBufferCount(2);
            data.EnsureDataType(ArrowTypeId.List);

            Values = values ?? ArrowArrayFactory.BuildArray(data.Children[0]);
        }
```

Reply:
> If users are confused by the values ​argument, how about adding this constructor as a new private constructor (without using a default value) and making both existing constructors call it.

Since `ArrayData` also has the values buffer in it, it would be duplicated with the `IArrowArray values` parameter. I think it is confusing for this to be a public constructor. I like your idea of making this new constructor private and having the 2 public constructors calling it.

Reply:
Thank you.
I'll add a new private constructor.